indico/modules/events/contributions/controllers/management.py

```diff
@@ -596,3 +612,45 @@ def _process(self):
             abstracts_settings.set(self.event_new, 'description_settings', form.data)
             return jsonify_data(flash=False)
         return jsonify_form(form)
+
+class RHCreateReferenceMixin:
+    """Common methods for RH class creating a ContibutionReference or SubContributionReference"""
+
+    def _checkParams(self):
+        self.reference_value = request.form['value']
+        reference_type_name = request.form['type']
+        self.reference_type = ReferenceType.find_one(name=reference_type_name)
```

Comment:
maybe make that one case insensitive since we also have the uniqueness on that column in a case insensitive way?

    ReferenceType.find_one(db.func.lower(ReferenceType.name) == reference_type_name.lower())